src/ui/control/geolocate_control.js

```diff
@@ -107,29 +241,118 @@ class GeolocateControl extends Evented {
             this._container);
         this._geolocateButton.type = 'button';
         this._geolocateButton.setAttribute('aria-label', 'Geolocate');
-        if (this.options.watchPosition) this._geolocateButton.setAttribute('aria-pressed', false);
+
+        if (this.options.trackUserLocation) {
+            this._geolocateButton.setAttribute('aria-pressed', false);
+            this._watchState = 'OFF';
+        }
+
+        // when showUserLocation is enabled, keep the Geolocate button disabled until the device location marker is setup on the map
+        if (this.options.showUserLocation) {
+            if (this.options.trackUserLocation) this._watchState = 'INITILIZE';
+
+            this._dotElement = DOM.create('div', 'mapboxgl-user-location-dot');
+
+            this._userLocationDotMarker = new Marker(this._dotElement, { offset: [-13, -13] });
```

Comment:
Will do.

On that note, I'm in favour of https://github.com/mapbox/mapbox-gl-js/issues/2900 which would eliminate offset here (and also make it much easier for people to override CSS styles to change the dot size for their app.